cli/nailgun-server/src/main/scala/com/rackspace/com/papi/components/checker/cli/NailgunServer.scala

```diff
@@ -1,116 +1,95 @@
+      head(s"$title v$implVersion")
 
-    val port = parser.option[Int] (List("p", "port"), "port",
-                                   s"The port for the server to listen to. Default: $PORT_DEFAULT")
+      help('h', "help").text("Display usage.")
 
-    val host = parser.option[String] (List("H", "host"), "host",
-                                      s"The host address for the server to listen to. Default: $HOST_DEFAULT")
+      version("version").text("Display version.")
 
-    val dontExitOnShutdown = parser.flag[Boolean] (List("e", "dont-exit-on-shutdown"),
-                                                   s"Dont exit the JVM when the server shutsdown.  Default: $DONT_EXIT_ON_SHUTDOWN")
+      opt[String]('d', "duration").text(s"The duration of time (in ISO 8601 format) before the server shuts down. Default: $DURATION_DEFAULT")
+        .foreach(x => duration = x)
 
-    val printVersion = parser.flag[Boolean] (List("v", "version"),
-                                             "Display version.")
+      opt[Int]('p', "port").text(s"The port for the server to listen to. Default: $PORT_DEFAULT")
+        .foreach(x => port = x)
 
-    val help = parser.flag[Boolean] (List("h", "help"),
-                                     "Display usage.")
+      opt[String]('H', "host").text(s"The host address for the server to listen to. Default: $HOST_DEFAULT")
+        .foreach(x => host = x)
 
-    try {
-      parser.parse(args)
+      opt[Unit]('e', "dont-exit-on-shutdown").text(s"Dont exit the JVM when the server shutsdown.  Default: $DONT_EXIT_ON_SHUTDOWN")
+        .foreach(_ => dontExitOnShutdown = true)
+    }
 
-      if (help.value.getOrElse(false)) {
-        parser.usage() // throws ArgotUsageException
-      }
 
-      if (printVersion.value.getOrElse(false)) {
-        System.err.println(s"$title v$version")
-        None
-      } else {
+    try {
+      if (parser.parse(args)) {
         //
         // Figure out the duration time
         //
-        val durationInMillis : Long = DatatypeFactory.newInstance.newDurationDayTime(duration.value.getOrElse(DURATION_DEFAULT)).getTimeInMillis(new Date)
+        val durationInMillis: Long = DatatypeFactory.newInstance.newDurationDayTime(duration).getTimeInMillis(new Date)
 
-        Some((host.value.getOrElse(HOST_DEFAULT),
-              port.value.getOrElse(PORT_DEFAULT),
-              durationInMillis,
-              !dontExitOnShutdown.value.getOrElse(DONT_EXIT_ON_SHUTDOWN)))
+        Some((host,
+          port,
+          durationInMillis,
+          !dontExitOnShutdown))
+
+      } else {
+        //failed to parse
+        None
       }
     } catch {
-      case e: ArgotUsageException => System.err.println(e.message)
-                                     None
-      case ia : IllegalArgumentException => System.err.println(ia.getMessage)
-                                            None
+      case ia: IllegalArgumentException => System.err.println(ia.getMessage)
```

Comment:
Is the `try` block still necessary here? I don't think an `IllegalArgumentException` will be thrown anymore? I think that is a remnant of the Argot parser. If I remember correctly, the Scopt parser will print its usage text and return `false`.

Reply:
Fair enough. I didn't look too closely at what was going on here, it just seemed odd to me that we have the conditional to handle the Scopt return nested in a try/catch block. 

Looking at it more closely, it seems that `IllegalArgumentException` would be thrown if the user provides a bad duration format. What is interesting is that line 75 can also `throw` a  `DatatypeConfigurationException` or `UnsupportedOperationException` if the user's environment is improperly set up, but neither of those exceptions are handled. So handling the `IllegalArgumentException` prints the exception message for us and gracefully exits, while the two unhandled exceptions would likely lead to a stacktrace being printed and an ungraceful exit. I suppose that behavior distinguishes between process argument issues and environment issues.

In that case, this seems mostly fine? Though it does seem a little odd to me to exit with an exit status of `0` when a command failed due to a bad argument being passed. However, that is consistent with how Scopt is handling that case.